README.md
<div align="center">
  <img src="https://user-images.githubusercontent.com/19170080/34070522-e15d32e2-e235-11e7-8af5-fa704cdcad56.png" />
</div>

# MUI-Datatables - Datatables for Material-UI

[![Build Status](https://travis-ci.org/gregnb/mui-datatables.svg?branch=master)](https://travis-ci.org/gregnb/mui-datatables)
[![Coverage Status](https://coveralls.io/repos/github/gregnb/mui-datatables/badge.svg?branch=master)](https://coveralls.io/github/gregnb/mui-datatables?branch=master)
[![dependencies Status](https://david-dm.org/gregnb/mui-datatables/status.svg)](https://david-dm.org/gregnb/mui-datatables)
[![npm version](https://badge.fury.io/js/mui-datatables.svg)](https://badge.fury.io/js/mui-datatables)

MUI-Datatables is a data tables component built on [Material-UI V1](https://www.material-ui-next.com).  It comes with features like filtering, view/hide columns, search, export to CSV download, printing, pagination, and sorting. On top of the ability to customize styling on most views, there are two responsive modes "stacked" and "scroll" for mobile/tablet devices. 

<div align="center">
	<img src="https://user-images.githubusercontent.com/19170080/34346996-b59a8b3a-e9cb-11e7-80d5-591aef3dc1f4.gif" />
</div>

## Install

`npm install mui-datatables --save`

## Demo

[![Edit react-to-print](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/wy2rl1nyzl)

## Usage


```js

import MUIDataTable from "mui-datatables";

const columns = ["Name", "Company", "City", "State"];

const data = [
 ["Joe James", "Test Corp", "Yonkers", "NY"],
 ["John Walsh", "Test Corp", "Hartford", "CT"],
 ["Bob Herm", "Test Corp", "Tampa", "FL"],
 ["James Houston", "Test Corp", "Dallas", "TX"],
];

const options = {
  filterType: 'checkbox',
};

<MUIDataTable 
  title={"Employee List"} 
  data={data} 
  columns={columns} 
  options={options} 
/>

```

## API


#### &lt;MUIDataTable />

The component accepts the following props:

|Name|Type|Description
|:--:|:-----|:-----|
|**`title`**|array|Title used to caption table
|**`columns`**|array|Columns used to describe table. Must be an array of simple strings
|**`data`**|array|Data used to describe table. Must be an array of simple strings
|**`options`**|object|Options used to describe table

#### Options:
|Name|Type|Default|Description
|:--:|:-----|:--|:-----|
|**`styles`**|object||Extend or override default styling
|**`filterType `**|string|'dropdown'|Choice of filtering view. Options are "checkbox" or "dropdown"
|**`pagination`**|boolean|true|Enable/disable pagination
|**`responsive`**|string|'stacked'|Enable/disable responsive table views. Options: 'stacked', 'scroll'
|**`rowsPerPage`**|number|10|Number of rows allowed per page
|**`rowsPerPageOptions`**|array|[10,15,20]|Options to provide in pagination for number of rows a user can select
|**`rowHover`**|boolean|true|Enable/disable hover style over rows
|**`sort`**|boolean|true|Show/hide sort icon from toolbar
|**`filter`**|boolean|true|Show/hide filter icon from toolbar
|**`search`**|boolean|true|Show/hide search icon from toolbar
|**`print`**|boolean|true|Show/hide print	 icon from toolbar
|**`download`**|boolean|true|Show/hide download icon from toolbar


## Customize Styling

In the options object, you have the ability to customize styling to your liking with the 'styles' property.  Here are the following sections you can customize:

#### Table of Contents

- [Table](#styletable)
- [Toolbar](#styletoolbar)
- [FilterList](#stylefilterlist)
- [Pagination](#stylepagination)
  
  
An example of how we would target FilterList would look like:

```js

const options = {
  filter: true,
  filterType: 'checkbox',
  styles: {
    filterList: {
      root: {
        color: "#FF0000"
      },
      chip: {
        color: "#FEFEF0"
      },
    },
  }
};

<MUIDataTable 
  title={"some title"} 
  data={data} 
  columns={columns} 
  options={options} 
/>

```

#### Styling Table
---
<a name="styletable"></a>


```js

const options = {
  styles: {
    table: {          
      head: {
        row: {
        },            
        cell: {
          root: {
          },
          sortLabel: {
          },
        }
      },
      body: {
        row: {
        },
        cell: {
          root: {
          }
        }
      },
    },
  }
};

```

#### Styling Toolbar
---
<a name="styletoolbar"></a>

```js

const options = {
  styles: {
    toolbar: {
      root: {},
      spacer: {
      },
      actions: {
      },
      titleRoot: {
      },
      titleText: {
      },
      icon: {
      },
      iconActive: {
      },
      search: {
      },
      searchIcon: {
      },
      searchText: {
      },
      clearIcon: {
      },
    },
  }
};

```

#### Styling FilterList
---
<a name="stylefilterlist"></a>



```js

const options = {
  styles: {
    filterList: {
      root: {
      },
      chip: {
      },
    },
  }
};

```

#### Styling Pagination
---
<a name="stylepagination"></a>

```js

const options = {
  styles: {  
    pagination: {
    }
  }
};

```


## License
The files included in this repository are licensed under the MIT license.


CHANGELOG.md
<a name="1.0.5"></a>
## 1.0.5 (2018-01-02)

Fix package.json issue

<a name="1.0.4"></a>
## 1.0.4 (2018-01-01)

* accessibility fixes + live announce area added to handle announcement of sorting columns and other f ([8dba3f7](https://github.com/gregnb/mui-datatables/commit/8dba3f7))
* added jsx-a11y to eslint. material-ui upgraded to 1.0.0.beta.26. now injecting MUIPopover with conta ([533a818](https://github.com/gregnb/mui-datatables/commit/533a818))
* added updatePosition to Popover after table row content updates ([baceb9f](https://github.com/gregnb/mui-datatables/commit/baceb9f))

<a name="1.0.3"></a>
## 1.0.3 (2017-12-30)

* additional code coverage. more clean up work ([e857b3f](https://github.com/gregnb/mui-datatables/commit/e857b3f))
* code coverage now at 91%. ready for first publish ([aedfdba](https://github.com/gregnb/mui-datatables/commit/aedfdba))
* created a better dataset for example demo. updated README ([edf4ce1](https://github.com/gregnb/mui-datatables/commit/edf4ce1))
* fix responsive 'stacked' view styling issues ([0debb9b](https://github.com/gregnb/mui-datatables/commit/0debb9b))

<a name="1.0.2"></a>
## 1.0.2 (2017-12-27)

* fix rollup issue. move version to 1.0.2 ([880983e](https://github.com/gregnb/mui-datatables/commit/880983e))
* Initial commit ([e09d24f](https://github.com/gregnb/mui-datatables/commit/e09d24f))
* keypress document listen event for ESCAPE on toolbar search ([bc6fc8d](https://github.com/gregnb/mui-datatables/commit/bc6fc8d))
* more unit test coverage. additional code cleanup ([94cad45](https://github.com/gregnb/mui-datatables/commit/94cad45))
* move to version 1.0.1 ([70b14c4](https://github.com/gregnb/mui-datatables/commit/70b14c4))
* ran prettier for cleanup ([f6ba65e](https://github.com/gregnb/mui-datatables/commit/f6ba65e))
* removed withDataStyles. replaced it with a new implementation: <DataStyles> ([b08a20b](https://github.com/gregnb/mui-datatables/commit/b08a20b))
* set prettier on test/ ([7799301](https://github.com/gregnb/mui-datatables/commit/7799301))
* swap SVG arrows for header columns. aria-labels for pagination ([128d4b3](https://github.com/gregnb/mui-datatables/commit/128d4b3))
* update codesandbox link ([fa8d33d](https://github.com/gregnb/mui-datatables/commit/fa8d33d))
* update README. add issue template ([cb7689b](https://github.com/gregnb/mui-datatables/commit/cb7689b))
* update unit tests to reflect changed caused by withDataStyles removal ([e60a00d](https://github.com/gregnb/mui-datatables/commit/e60a00d))





example/index.js
import React from "react";
import ReactDOM from "react-dom";
import MUIDataTable from "../src/";

class Example extends React.Component {

  render() {

    const columns = ["Name", "Title", "Location", "Age", "Salary"];

    const data = [
      ["Gabby George", "Business Analyst", "Minneapolis", 30, "$100,000"],
      ["Aiden Lloyd", "Business Consultant", "Dallas",  55, "$200,000"],
      ["Jaden Collins", "Attorney", "Santa Ana", 27, "$500,000"],
      ["Franky Rees", "Business Analyst", "St. Petersburg", 22, "$50,000"],
      ["Aaren Rose", "Business Consultant", "Toledo", 28, "$75,000"],
      ["Blake Duncan", "Business Management Analyst", "San Diego", 65, "$94,000"],
      ["Frankie Parry", "Agency Legal Counsel", "Jacksonville", 71, "$210,000"],
      ["Lane Wilson", "Commercial Specialist", "Omaha", 19, "$65,000"],
      ["Robin Duncan", "Business Analyst", "Los Angeles", 20, "$77,000"],
      ["Mel Brooks", "Business Consultant", "Oklahoma City", 37, "$135,000"],
      ["Harper White", "Attorney", "Pittsburgh", 52, "$420,000"],
      ["Kris Humphrey", "Agency Legal Counsel", "Laredo", 30, "$150,000"],
      ["Frankie Long", "Industrial Analyst", "Austin", 31, "$170,000"],
      ["Brynn Robbins", "Business Analyst", "Norfolk", 22, "$90,000"],
      ["Justice Mann", "Business Consultant", "Chicago", 24, "$133,000"],
      ["Addison Navarro", "Business Management Analyst", "New York", 50, "$295,000"],
      ["Jesse Welch", "Agency Legal Counsel", "Seattle", 28, "$200,000"],
      ["Eli Mejia", "Commercial Specialist", "Long Beach", 65, "$400,000"],
      ["Gene Leblanc", "Industrial Analyst", "Hartford", 34, "$110,000"],
      ["Danny Leon", "Computer Scientist", "Newark", 60, "$220,000"],
      ["Lane Lee", "Corporate Counselor", "Cincinnati", 52, "$180,000"],
      ["Jesse Hall", "Business Analyst", "Baltimore", 44, "$99,000"],
      ["Danni Hudson", "Agency Legal Counsel", "Tampa", 37, "$90,000"],
      ["Terry Macdonald", "Commercial Specialist", "Miami", 39, "$140,000"],
      ["Justice Mccarthy", "Attorney", "Tucson", 26, "$330,000"],
      ["Silver Carey", "Computer Scientist", "Memphis", 47, "$250,000" ],
      ["Franky Miles", "Industrial Analyst", "Buffalo", 49, "$190,000"],
      ["Glen Nixon", "Corporate Counselor", "Arlington", 44, "$80,000"],
      ["Gabby Strickland", "Business Process Consultant", "Scottsdale", 26, "$45,000"],
      ["Mason Ray", "Computer Scientist", "San Francisco", 39, "$142,000"]
    ];


    const options = {
      filter: true,
      filterType: 'dropdown',
      responsive: 'stacked',
      styles: {
        table: {
          
          main: {
            root: {
            },
            responsiveScroll: {

            },
            responsiveStacked: {
            }
          },

          head: {

            row: {
            },
            
            cell: {
              root: {
              },
              sortLabel: {
              },
            }

          },
          body: {

            main: {

            },

            row: {
              root: {
              }
            },
            cell: {
              root: {
              }
            }

          },

        },
        toolbar: {
          root: {},
          spacer: {
          },
          actions: {
          },
          titleRoot: {
          },
          titleText: {
          },
          icon: {
            "&:hover": {
            },
          },
          iconActive: {
          },
          searchIcon: {
          },
        },
        filterView: {
        },
        filterList: {
          root: {
            color: "#000"
          },
          chip: {
            color: "#FEFEF0"
          },
        },
        pagination: {
          root: {
            border: "solid 1px #000"
          }
        },
        viewColumns: {

        },
        search: {
          main: {
          },
          searchIcon: {
          },
          searchText: {
          },
          clearIcon: {
          },          
        }
      }
    };

    return (
      <MUIDataTable title={"ACME Employee list"} data={data} columns={columns} options={options} />
    );

  }
}

ReactDOM.render(<Example />, document.getElementById("app-root"));
